scripts/fortran_tools/parse_fortran_file.py

```diff
@@ -348,112 +348,113 @@ def read_file(filename, preproc_defs=None, logger=None):
     preproc_status = PreprocStack()
     if not os.path.exists(filename):
         raise IOError("read_file: file, '{}', does not exist".format(filename))
-    else:
-        # We need special rules for fixed-form source
-        fixed_form = filename[-2:].lower() == '.f'
-        # Read all lines of the file at once
-        with open(filename, 'r') as file:
-            file_lines = file.readlines()
-            for index, line in enumerate(file_lines):
-                file_lines[index] = line.rstrip('\n').rstrip()
-            # End for
-        # End with
-        # create a parse object and context for this file
-        pobj = ParseObject(filename, file_lines)
-        continue_col = -1 # Active continue column
-        in_schar = False # Single quote character context
-        in_dchar = False # Double quote character context
-        prev_line = None
-        prev_line_num = -1
-        curr_line, curr_line_num = pobj.curr_line()
-        while curr_line is not None:
-            # Skip empty lines and comment-only lines
-            skip_line = False
-            if len(curr_line.strip()) == 0:
-                skip_line = True
-            elif (fixed_form and
-                  (_FIXED_COMMENT_RE.match(curr_line) is not None)):
-                skip_line = True
-            elif curr_line.lstrip()[0] == '!':
-                skip_line = True
-            # End if
-            if skip_line:
-                curr_line, curr_line_num = pobj.next_line()
-                continue
-            # End if
-            # Handle preproc issues
-            if preproc_status.process_line(curr_line, preproc_defs, pobj, logger):
+    # end if
+    # We need special rules for fixed-form source
+    fixed_form = filename[-2:].lower() == '.f'
```

Comment:
Unfortunately not all Fortran files in the current ccpp-physics follow the rule. Some have `.f` but have special compiler flags to treat them as free-form files. I guess we'll have to rename those, there is no way for `cap_gen.py` to know these flags. It's better anyway.